lib/galaxy/tools/toolbox/watcher.py

```diff
@@ -98,7 +98,7 @@ def shutdown(self):
 
     def check(self):
         """Check for changes in self.paths or self.cache and call the event handler."""
-        hashes = {key: None for key in self.paths.keys()}
+        hashes = {key: None for key in list(self.paths.keys())}
```

Comment:
Can't we just drop this line?

Reply:
```suggestion
        hashes = {}
```
? That should work, yes.